### ISSUE TITLE
awful.placement.no_overlap is not honored by ruled clients

### ISSUE THREAD

**Issue by sudo-nice** [awesomeWM/awesome#1511]
Output of `awesome --version`:
```
awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✔
 • LGI version: 0.9.1
```
**How to reproduce the issue:**

*Let's assume we use the default rc.lua*

1. Edit the default rc.lua to add the rule:
```
{ rule = { class = "XTerm" },
	properties = { screen = 1, tag = "2", switchtotag = false } },
```
2. Make sure you are on the tag 1.
3. Spawn xterm twice or more times, to have the clients placed on the tag 2.
4. Switch to the tag 2.

**Actual result:**

The clients are overlapped.

**Expected result:**

The clients are placed side by side.

**Notes:**

The problem doesn't happen for clients spawned on the tag 1 and moved to the tag 2 manually.

**Comment by psychon** [comment]
`awful.placement.no_overlap` only considers visible clients (`awful.client.visible(scr)`), so that's why this behaviour occurs. I have no suggestions what to do about it... @Elv13 ?

**Comment by Elv13** [comment]
Yes, I was looking at the same time. This was always an issue. This code is still unported to the new placement API conventions. There is an issue for that too

**Comment by psychon** [comment]
Removed the milestone since it is not a regression and I think it shouldn't block the release. (My first thought was to move to the `v4.3` milestone, but that would just be lying to myself.)
